### PR TITLE
DACT-1657: Skip address validation if sameAs flags are set

### PR DESCRIPTION
**Pull request by anona-w**
https://companieshouse.atlassian.net/browse/DACT-1657
Skipping duplicate validation checks on linked addresses if SameAs flags are set, as the linked to address will have already been validated.

### CODE REVIEW

**Review by mdesaikns — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java [hunk 1]
```diff
@@ -102,7 +108,7 @@ public void validateOptionalDtoFields(HttpServletRequest request, List<ApiError>
         validateMiddleNames(request, errorList, dto);
         validateFormerNames(request, errorList, dto);
         validateOccupation(request, errorList, dto);
-        if(dto.getResidentialAddress() != null){
+        if(dto.getResidentialAddress() != null && Boolean.FALSE.equals(dto.getIsHomeAddressSameAsServiceAddress())) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java:111]: Is there a similar validation for the optional correspondence address fields?

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java:111]: No, seems to be some inconsistency in approach to validating addresses.

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java [hunk 2]
```diff
@@ -89,11 +89,17 @@ public void validateRequiredDtoFields(HttpServletRequest request, List<ApiError>
         validateNationality2(request, errorList, dto);
         validateNationality3(request, errorList, dto);
         validateNationalityLength(request, errorList, dto);
-        validateRequiredResidentialAddressFields(request, errorList, dto);
-        validateCorrespondenceAddressFields(request, errorList, dto);
         validateAppointmentDate(request, errorList, dto);
         validateProtectedDetails(request, errorList, dto);
         validateConsentToAct(request, errorList, dto);
+
+        if (Boolean.FALSE.equals(dto.getIsHomeAddressSameAsServiceAddress())) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java:96]: Should this not validate if the flag is undefined? Defaulting to false. So !Boolean.True.equals()

Comment (anona-w) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java]: You're correct Andre - I'd previously had a null check as well and had removed it where I shouldn't have, but your suggestion is cleaner anyway. Have made that change and re-tested and the null case works as it should now. Nice catch!

**Review by AndreKramer7654 — APPROVED** [review]

**Review by AndreKramer7654 — APPROVED** [review]

**Review by clewis1 — APPROVED** [review]
Looks good.

**Pull request merged**